cmd/krew/cmd/internal/setup_check_test.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"sigs.k8s.io/krew/internal/environment"
+	"sigs.k8s.io/krew/internal/testutil"
+)
+
+func TestIsBinDirInPATH_firstRun(t *testing.T) {
+	tempDir, cleanup := testutil.NewTempDir(t)
+	defer cleanup()
+
+	paths := environment.NewPaths(tempDir.Path("does-not-exist"))
+	res := IsBinDirInPATH(paths)
+	if res == false {
+		t.Errorf("Expected positive result on first run")
+	}
+}
+
+func TestIsBinDirInPATH_secondRun(t *testing.T) {
+	tempDir, cleanup := testutil.NewTempDir(t)
+	defer cleanup()
+	if err := os.Mkdir(tempDir.Path("exists"), os.ModePerm); err != nil {
+		t.Fatal(err)
+	}
```

Comment:
can we just pass tempDir as path, instead of creating a dir called "exists"?

Reply:
Sure. I wanted to make it more explicit that the directory is there vs. the non-existent case. But I guess this will do as well.